Lively/src/components/session/TeamEntryForm.tsx

```diff
@@ -0,0 +1,166 @@
+import * as React from 'react';
+import './TeamEntry.css';
+import { handleTokens } from '../../util';
+import { Submit, Loading } from '../wrappers/Buttons';
+import { validateEmail } from '../../util';
+import { MutationFunc } from 'react-apollo';
+
+interface State {
+    email: string;
+    password: string;
+    loading: boolean;
+    badEmail: boolean; 
+    badPassword: boolean;
+    shouldFocus: boolean;
+}
+
+type MutationPayload = {
+    teamLogin: {
+        ok: boolean;
+        token?: string;
+        refreshToken?: string;
+    }
+};
+  
+type MutationInput = {
+    url: string;
+    password: string; 
+    email: string
+ };
+  
+interface Props {
+    url: string;
+    setError: (error: boolean) => void; 
+    teamLogin: MutationFunc<MutationPayload, MutationInput>;
+}
+
+class TeamEntryForm extends React.Component<Props, State> {
+    state = {
+        email: '', 
+        password: '', 
+        loading: false,
+        badEmail: false,
+        badPassword: false,
+        shouldFocus: false,
+    };
+
+    badInputStyle = {border: '1px solid #d72b3f', background: '#fbeaec'};
+
+    handleChange = (e: React.FormEvent<HTMLInputElement>) => {
+        this.setState({
+            [e.currentTarget.name as any]: e.currentTarget.value // tslint:disable-line
+        });
+    }
+
+    checkBadFields = (submitting = false) => {
```

Comment:
there's commonly used technical jargon for certain kinds of things. in actuality this function validates a form, it should be named accordingly